File: Ejercicios C/14.Variables.cpp
#include <iostream>
#include <typeinfo>  // Para obtener el tipo de la variable
#include<math.h>
using namespace std;

int main() {
    // Declaramos las variables
    int a = 10, b = 4;
    int c;  // Variable para el resultado

    // Multiplicación
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operacion es: * " << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a * b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;  // Mostrar el tipo de c

    // División
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operacion es: / " << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a / b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;

    // Suma
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operacion es: + " << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a + b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;

    // Resta
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operacion es: - " << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a - b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;

    // Potenciación (usamos el `pow` para potencia en C++)
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operacion es: ** " << endl;
    cout << "La segunda variable es: " << b << endl;
    c = static_cast<int>(pow(a, b));  // Convertimos el resultado a int
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;

    // Porcentaje
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de la operacion es: % " << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a % b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;

    return 0;
}

File: Ejercicios PHP/14.Variables.php
<?php
// Aplicando variables.
$a = 10;
$b = 4;

// Multiplicación:
echo 'La primera variable es: ' . $a . "\n";
echo 'El signo de la operación es: * ' . "\n";
echo 'La segunda variable es: ' . $b . "\n";
$c = ($a * $b);
echo 'El resultado es: ' . $c . "\n";
echo 'El tipo de dato es: ' . gettype($c) . "\n\n";

// División:
echo 'La primera variable es: ' . $a . "\n";
echo 'El signo de la operación es: / ' . "\n";
$c = ($a / $b);
echo 'El resultado es: ' . $c . "\n";
echo 'El tipo de dato es: ' . gettype($c) . "\n\n";

// Suma:
echo 'La primera variable es: ' . $a . "\n";
echo 'El signo de la operación es: + ' . "\n";
echo 'La segunda variable es: ' . $b . "\n";
$c = ($a + $b);
echo 'El resultado es: ' . $c . "\n";
echo 'El tipo de dato es: ' . gettype($c) . "\n\n";

// Resta:
echo 'La primera variable es: ' . $a . "\n";
echo 'El signo de la operación es: - ' . "\n";
echo 'La segunda variable es: ' . $b . "\n";
$c = ($a - $b);
echo 'El resultado es: ' . $c . "\n";
echo 'El tipo de dato es: ' . gettype($c) . "\n\n";

// Potenciación:
echo 'La primera variable es: ' . $a . "\n";
echo 'El signo de la operación es: ** ' . "\n";
echo 'La segunda variable es: ' . $b . "\n";
$c = ($a ** $b);
echo 'El resultado es: ' . $c . "\n";
echo 'El tipo de dato es: ' . gettype($c) . "\n\n";

// Porcentaje (módulo):
echo 'La primera variable es: ' . $a . "\n";
echo 'El signo de la operación es: % ' . "\n";
echo 'La segunda variable es: ' . $b . "\n";
$c = ($a % $b);
echo 'El resultado es: ' . $c . "\n";
echo 'El tipo de dato es: ' . gettype($c) . "\n\n";

// Para ejecutar en terminal:
// php "C:\Sena\Analisis y desarrollo de software\Ficha 3066478 segundo trimestre\Ejercicios PHP\14.Variables.php"
?>
